src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -126,21 +126,20 @@ private function addTransitions(array $transitions)
     private function findEdges(Definition $definition)
     {
         $dotEdges = array();
+        $getEdge = function ($from, $to, $direction) use (&$dotEdges) {
+            $dotEdges[$from.$to.$direction] = array(
```

Comment:
the thing is that your dumping of the StateMachine graph does not represent the workflow being defined by the state machine, as you represent a single transition with multiple input and output instead of several transitions (the fact that the DI config of the StateMachine allows to define these transitions in a shorted way by grouping them does *not* mean they are the same transition in the StateMachine)
Your dumping is lying about the actual workflow being run.
The bug in the GraphvizDumper currently is that it uses transition names to identify them in the dot file, while they are not unique identifiers anymore (they were at the time the dumper was written).

Btw, this is why I said we should have another dumper being tailored only at state machines, and dumping them into a graph being more readable. This graph would represent transitions as edges on the graph (which is not possible in generic workflows, as transitions can have multiple input and output). But I would make this another dumper, or an option in this dumper, because dumping the StateMachine as a generic workflow graph can be useful too.